Zend/zend_inheritance.c

```diff
@@ -1200,6 +1200,8 @@ static zend_always_inline void do_inherit_method(zend_string *key, zend_function
 				func, func->common.scope, parent, parent->common.scope, ce, child,
 				/* check_visibility */ 1);
 		}
+
+		func->common.fn_flags &= ~ZEND_ACC_OVERRIDE;
```

Comment:
This should only remove the flags on methods that belong to the class itself.
```c
if (func->common.scope == ce) {
	func->common.fn_flags &= ~ZEND_ACC_OVERRIDE;
}
```
Otherwise we can:
1. Modify methods from parent classes that are in shm, which may lead to race conditions
2. The above would hide errors of the parent class having the override attribute on methods that are not overridden

Test case:
```php
interface IFace {
    function test();
}

class Base {
    #[Override]
    function test() {}
}

class Test extends Base implements IFace {}
```

Reply:
> Do interface methods count as overridden?

Yes they should. The attribute is modeled after Java's `@Override` annotation and intended as an assistance to the reader whether the method is specific to the class or required due to some external constraints. It allows the user to detect methods that are effectively dead code, because they have been removed from the interface and thus will never be called again.

Reply:
Btw I think this would fit better into `do_inheritance_check_on_method_ex`.